expand/expand.go

```diff
@@ -57,6 +58,12 @@ type Config struct {
 	// Use ioutil.ReadDir to use the filesystem directly.
 	ReadDir func(string) ([]os.FileInfo, error)
 
+	// Stat is used as an optimization, to get an os.FileInfo for a single
+	// file. Only needed for Fields, and then only if ReadDir is set. If Stat
+	// is not set, ReadDir is called on name's parent directory. Use os.Stat to
+	// use the filesystem directly.
```

Comment:
I'm not sure if I agree with the docs. If ReadDir is set, Stat is about optimizing, but if ReadDir isn't set, it's still about sandboxing.

Reply:
If `ReadDir` isn't set, `Stat` is never called.  It's only called in `Config.glob`, and that's only called if `ReadDir != nil`.

Reply:
You're right, I had forgotten that detail.